### ISSUE TITLE
Save file suddenly disabled

### ISSUE THREAD

**Issue by Seemply** [texstudio-org/texstudio#2094]
### Environment
- TeXstudio: 4.2.0 as well as 4.2.1
- Qt: 6.2.1, kompiliert mit Qt 6.2.1 R
- OS: Windows 10
- TeX distribution: MiKTeX-pdfTeX 4.10 (MiKTeX 21.12.10)

### Expected behavior

After changing text in the editor TexStudio should keep the option to save enabled (both the 'save' button and the menu entry).

### Actual behavior

'Save' button as well as menu entry are suddenly disabled. The shortcut CTRL-S doesn't work either. 

![S1643630254-horz](https://user-images.githubusercontent.com/98743179/151790421-bdcdf3cd-d433-4283-9bf1-1724930f054d.jpg)

Option 'save as' works as usual, though. Restart of the program solves the problem for a while until it happens again.

### How to reproduce

I can't reproduce it deliberately. This problem started with the last version 4.20 I believe. 

There is another 'bug' since then, most likely unrelated: When doing 'Undo' in the editor sometimes 'random' text appears in the editor window.

**Comment by sunderme** [comment]
some more hints needed, i.e. reproducible example.

**Comment by Seemply** [comment]
I think I've got it pretty much isolated now. Hope the animated gif works:

![UA](https://user-images.githubusercontent.com/98743179/151861358-485ca92e-322a-4e9f-96c2-801655022357.gif)

What I did:

1. Take a new file with just a few lines plus a new user environment
2. remove \end
3. Compile with PDFLatex, which works fine
4. delete \end{buggy} (using the great placeholder idea)
5. Another compilation leads to an error message as expected
6. Save the file via CTRL-S.
7. Can't save file now. 

Further editor input won't change this. I have to go to 'save as' to 'save (_my_) as' Also CTRL-S or going through the menu doesn't work either. Could be a placeholder issue - as there was one in January?
